src/main/java/io/kestra/plugin/notion/NotionConnection.java

```diff
@@ -0,0 +1,221 @@
+package io.kestra.plugin.notion;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.http.HttpRequest;
+import io.kestra.core.http.HttpResponse;
+import io.kestra.core.http.client.HttpClient;
+import io.kestra.core.http.client.configurations.HttpConfiguration;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.serializers.JacksonMapper;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import reactor.core.publisher.Flux;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class NotionConnection extends Task {
+
+    protected final static ObjectMapper mapper = JacksonMapper.ofJson(false);
+
+    public static final String NOTION_API_URL = "https://api.notion.com";
+
+    public static final String NOTION_API_VERSION = "2022-06-28";
```

Comment:
Maybe this is something that should be settable by the user?

Reply:
Let's keep it like this for now if that's ok to you :) Waiting for feedback before making these ones public